### PR TITLE
Refactor: Request Response Refactor

### PR DESCRIPTION
**Pull request by amitksingh1490**

### CODE REVIEW

**Comment by tusharmath** [comment]
Try to have minimal changes in the examples @amitksingh1490

File: zio-http-benchmarks/src/main/scala/zhttp.benchmarks/HttpRouteTextPerf.scala [hunk 1]
```diff
@@ -13,11 +13,12 @@ class HttpRouteTextPerf {
...
 
-  private val res                        = Response.text("HELLO WORLD")
-  private val app: HttpApp[Any, Nothing] = HttpApp.text("HELLO WORLD")
-  private val req: Request               = Request(Method.GET -> URL(Root))
-  private val httpProgram                = ZIO.foreach_(0 to 1000) { _ => app.execute(req).evaluate.asEffect }
-  private val UIOProgram                 = ZIO.foreach_(0 to 1000) { _ => UIO(res) }
+  private val res                                 = Response.text("HELLO WORLD")
+  private val app: HttpApp[Any, Nothing]          = HttpApp.text("HELLO WORLD")
+  private val req: Request[Any, Nothing, Nothing] =
+    Request(Method.GET -> URL(Root), Nil, HttpData.empty).asInstanceOf[Request[Any, Nothing, Nothing]]
```
Comment (amitksingh1490) [inline zio-http-benchmarks/src/main/scala/zhttp.benchmarks/HttpRouteTextPerf.scala:19]: Discuss this

File: zio-http/src/main/scala/zhttp/service/server/ServerRequestHandler.scala [hunk 2]
```diff
@@ -60,32 +56,36 @@ final case class ServerRequestHandler[R](
    */
   override def channelRead0(ctx: JChannelHandlerContext, jReq: JFullHttpRequest): Unit = {
     executeAsync(ctx, jReq) {
-      case res @ Response.HttpResponse(_, _, content) =>
-        ctx.write(encodeResponse(jReq.protocolVersion(), res), ctx.channel().voidPromise())
+      case res @ Response.Default(_, _, content) =>
+        ctx.write(
+          encodeResponse(jReq.protocolVersion(), res.asInstanceOf[Response[Any, Nothing, Opaque]]),
```
Comment (amitksingh1490) [inline zio-http/src/main/scala/zhttp/service/server/ServerRequestHandler.scala:61]: Discuss this

File: example/src/main/scala/SimpleClient.scala [hunk 3]
```diff
@@ -11,9 +11,8 @@ object SimpleClient extends App {
     res <- Client.request(url, headers)
     _   <- console.putStrLn {
       res.content match {
-        case HttpData.CompleteData(data) => data.map(_.toChar).mkString
-        case HttpData.StreamData(_)      => "<Chunked>"
-        case HttpData.Empty              => ""
+        case HttpData.CompleteContent(data) => data.map(_.toChar).mkString
```
Comment (amitksingh1490) [inline example/src/main/scala/SimpleClient.scala:14]: Discuss

**Pull request closed without merging**